pyrefly/lib/binding/scope.rs

```diff
@@ -807,10 +808,42 @@ impl Scopes {
         None
     }
 
-    pub fn get_flow_style(&self, name: &Name) -> &FlowStyle {
+    fn get_static_info(&self, name: &Name, should_skip_current_scope: bool) -> Option<&StaticInfo> {
+        let name = Hashed::new(name);
+        let mut iter = self.iter_rev();
+        if should_skip_current_scope {
+            iter.next();
+        }
+        for scope in iter {
+            if let Some(info) = scope.stat.0.get_hashed(name) {
+                return Some(info);
+            }
+        }
+        None
+    }
+
+    pub fn get_flow_style(&self, name: &Name, usage: &mut Usage) -> &FlowStyle {
```

Comment:
We probably should put a doc comment here, and explicitly call out that it's not really just getting the flow style anymore, it's doing additional checks of static info (with the appropriate semantic information) to track initialization except in type annotations